src/Components/test/E2ETest/Tests/WebAssemblyConfigurationHostedTest.cs

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using BasicTestApp;
+using Microsoft.AspNetCore.Components.E2ETest;
+using Microsoft.AspNetCore.Components.E2ETest.Infrastructure;
+using Microsoft.AspNetCore.Components.E2ETest.Infrastructure.ServerFixtures;
+using Microsoft.AspNetCore.E2ETesting;
+using OpenQA.Selenium;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.AspNetCore.Components.E2ETests.Tests
+{
+    public class WebAssemblyConfigurationHostedTest : ServerTestBase<BasicTestAppServerSiteFixture<TestServer.ClientStartup>>
+    {
+        private IWebElement _appElement;
+
+        public WebAssemblyConfigurationHostedTest(
+             BrowserFixture browserFixture,
+             BasicTestAppServerSiteFixture<TestServer.ClientStartup> serverFixture,
+             ITestOutputHelper output) :
+             base(browserFixture, serverFixture, output)
+        {
+        }
+
+        protected override void InitializeAsyncCore()
+        {
+            base.InitializeAsyncCore();
+
+            Navigate(ServerPathBase, noReload: false);
+            _appElement = Browser.MountTestComponent<ConfigurationComponent>();
+        }
+
+        [Fact]
+        public void WebAssemblyConfiguration_Works()
+        {
+            // Verify values from the default 'appsettings.json' are read.
+            Browser.Equal("Default key1-value", () => _appElement.FindElement(By.Id("key1")).Text);
+
+            // Verify values overriden by an environment specific 'appsettings.$(Environment).json are read
+            Assert.Equal("Prod key2-value", _appElement.FindElement(By.Id("key2")).Text);
+
+            // Lastly for sanity, make sure values specified in an environment specific 'appsettings.$(Environment).json are read
+            Assert.Equal("Prod key3-value", _appElement.FindElement(By.Id("key3")).Text);
```

Comment:
Are these assertions going to fail when we run the E2E tests locally? Won't they run in `Development` environment by default because `TestServer` has `"ASPNETCORE_ENVIRONMENT": "Development"` in its `launchSettings.json`?

Reply:
`launchSettings.json` is a feature of `dotnet run` \ VS. It doesn't get used the way we host our apps in the E2E tests, so this works fine.

Reply:
Good point